src/main/java/com/google/devtools/build/lib/query2/cquery/TransitionResolver.java

```diff
@@ -0,0 +1,168 @@
+     *
+     * If a target is depended on via two attributes, separate ResolvedTransitions should be used,
+     * rather than combining the two into a single ResolvedTransition with multiple options.
+     *
+     * If no transition was applied to an attribute, this collection will be empty.
+     */
+    abstract Collection<BuildOptions> options();
+
+    /**
+     * The name of the attribute via which the dependency was requested.
+     */
+    abstract String attributeName();
+
+    /**
+     * The name of the transition applied to the attribute.
+     */
+    abstract String transitionName();
+  }
+
+  private final ExtendedEventHandler eventHandler;
+  private final DependencyResolver dependencyResolver;
+  private final ConfiguredTargetAccessor accessor;
+  private final CqueryThreadsafeCallback cqueryThreadsafeCallback;
+  @Nullable private final TransitionFactory<RuleTransitionData> trimmingTransitionFactory;
+
+  public TransitionResolver(
+      ExtendedEventHandler eventHandler,
+      DependencyResolver dependencyResolver,
+      ConfiguredTargetAccessor accessor,
+      CqueryThreadsafeCallback cqueryThreadsafeCallback,
+      @Nullable TransitionFactory<RuleTransitionData> trimmingTransitionFactory) {
+    this.eventHandler = eventHandler;
+    this.dependencyResolver = dependencyResolver;
+    this.accessor = accessor;
+    this.cqueryThreadsafeCallback = cqueryThreadsafeCallback;
+    this.trimmingTransitionFactory = trimmingTransitionFactory;
+  }
+
+  /**
+   * Return the set of dependencies of a KeyedConfiguredTarget, including information about the
+   * configuration transitions applied to the dependencies.
+   *
+   * @see ResolvedTransition for more details.
+   *
+   * @param keyedConfiguredTarget the configured target whose dependencies are being looked up.
+   */
+  public LinkedHashSet<ResolvedTransition> dependencies(KeyedConfiguredTarget keyedConfiguredTarget) throws DependencyResolver.Failure, InconsistentAspectOrderException, InterruptedException {
+    LinkedHashSet<ResolvedTransition> resolved = new LinkedHashSet<>();
+
+    if (!(keyedConfiguredTarget.getConfiguredTarget() instanceof RuleConfiguredTarget)) {
+      return resolved;
+    }
+
+    Target target = accessor.getTarget(keyedConfiguredTarget);
+    BuildConfigurationValue config =
+        cqueryThreadsafeCallback.getConfiguration(keyedConfiguredTarget.getConfigurationKey());
+
+    OrderedSetMultimap<DependencyKind, DependencyKey> deps;
+    ImmutableMap<Label, ConfigMatchingProvider> configConditions =
+        keyedConfiguredTarget.getConfigConditions();
+
+    // Get a ToolchainContext to use for dependency resolution.
+    ToolchainCollection<ToolchainContext> toolchainContexts =
+        accessor.getToolchainContexts(target, config);
+    // We don't actually use fromOptions in our implementation of
+    // DependencyResolver but passing to avoid passing a null and since we have the information
+    // anyway.
+    deps =
```

Comment:
would it be clearer to do `OrderedSetMultimap<DependencyKind, DependencyKey> deps = ...` and delete line 117?

Reply:
Absolutely it would! I think originally this code before I moved it was in a try-catch, hence the separation :) Fixed